### PR TITLE
[avce] adaptive P-pyramid fix

### PR DESCRIPTION
**Pull request by dkurguzi**
Issue: VSMGWL-31306
Test: manually

### CODE REVIEW

**Review by lakulako — APPROVED** [review]
Is it backmerge?

**Comment by dkurguzi** [comment]
> Is it backmerge?

thank you for reveiw, yes it is backmerge

File: _studio/mfx_lib/encode_hw/h264/include/mfx_h264_encode_hw_utils.h [hunk 1]
```diff
@@ -2891,7 +2891,8 @@ class H264EncTools
         mfxStatus BuildPPyr(
             DdiTask & task,
             mfxU32 pyrWidth,
-            bool bLastFrameUsing);
+            bool bLastFrameUsing,
+             bool bResetPyr);
```
Comment (pgribov) [inline _studio/mfx_lib/encode_hw/h264/include/mfx_h264_encode_hw_utils.h:2895]: extra space

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp [hunk 2]
```diff
@@ -6060,7 +6060,7 @@ void MfxHwH264Encode::SetDefaults(
         else
             par.mfx.NumRefFrame = std::min({nrfDefault, nrfMaxByLevel, nrfMaxByCaps});
 
-        if (IsOn(par.mfx.LowPower))
+         if (IsOn(par.mfx.LowPower) && (extOpt3->PRefType != MFX_P_REF_PYRAMID))
```
Comment (pgribov) [inline _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp:6063]: extra space

**Review by pgribov — APPROVED** [review]

File: _studio/mfx_lib/encode_hw/h264/include/mfx_h264_encode_hw_utils.h (hunk 1 above)
Comment (dkurguzi) [reply _studio/mfx_lib/encode_hw/h264/include/mfx_h264_encode_hw_utils.h]: fixed, thank you

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp (hunk 2 above)
Comment (dkurguzi) [reply _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: fixed, thank you

**Review by takulako — APPROVED** [review]

**Review by aleonenk — APPROVED** [review]

**Pull request merged**